### PR TITLE
storage: Use TypeAheadSelect instead of our implementation of ComboBox

### PR DESCRIPTION
**Pull request by marusak**
While fixing rule in #12854, I got to this component about which the linter said is not accessible and was right - you can just navigate into the input, but you cannot access the button to open the dropdown (once you open it by mouse you can use tab to move through these items but still cannot use enter to select this items).

With this component I can access it by keyboard, use arrows to navigate though items, use enter to select these items... so it is very nicely accessible by keyboard.

It is PF3 component - we use it on two another places, so I think it is fine to use this version. Or @KKoukiou do you have some idea about this component in PF3 vs PF4?

### CODE REVIEW

**Comment by KKoukiou** [comment]
> While fixing rule in #12854, I got to this component about which the linter said is not accessible and was right - you can just navigate into the input, but you cannot access the button to open the dropdown (once you open it by mouse you can use tab to move through these items but still cannot use enter to select this items).
> 
> With this component I can access it by keyboard, use arrows to navigate though items, use enter to select these items... so it is very nicely accessible by keyboard.
> 
> It is PF3 component - we use it on two another places, so I think it is fine to use this version. Or @KKoukiou do you have some idea about this component in PF3 vs PF4?

We are already using this component in other places, see pkg/machines/components/create-vm-dialog/createVmDialog.jsx
I suggest to stick to the PF3 variant in this PR, there were some issues with the PF4 which seem to be solved https://github.com/patternfly/patternfly-react/issues/2444

And I can migrate all occurances of this component in our code all at  once.

File: pkg/storaged/dialog.jsx [hunk 1]
```diff
@@ -214,10 +214,11 @@
 import cockpit from "cockpit";
 
 import React from "react";
-import { OverlayTrigger, Tooltip } from "patternfly-react";
+import { OverlayTrigger, Tooltip, TypeAheadSelect } from "patternfly-react";
 
 import { show_modal_dialog } from "cockpit-components-dialog.jsx";
 import { StatelessSelect, SelectEntry } from "cockpit-components-select.jsx";
+
```
Comment (KKoukiou) [inline pkg/storaged/dialog.jsx]: Not related newline change.

File: pkg/storaged/dialog.jsx [hunk 2]
```diff
@@ -537,8 +493,30 @@ export const ComboBox = (tag, title, options) => {
...
+                <TypeAheadSelect
+                    id="nfs-path-on-server"
+                    labelKey="path"
+                    placeholder=""
+                    paginate={false}
+                    onChange={value => change(value[0])}
+                    onInputChange={change}
+                    options={options.choices}
+                    disabled={options.disabled}
+                    onKeyDown={ev => { // Capture ESC event
```
Comment (KKoukiou) [inline pkg/storaged/dialog.jsx]: So I know I wrote this code for the file autocompletion but I was just testing it and it doesn't seem to work (neither in the file autocompletion component)
I meaning that hitting Esc will just close the dialog, which is not what we want here.

Comment (marusak) [reply pkg/storaged/dialog.jsx]: Ech yeah, I shamelessly stole it from file autocomplete component :D 
I prefer to fix it together outside of this PR

File: pkg/storaged/dialog.jsx (hunk 1 above)
Comment (marusak) [reply pkg/storaged/dialog.jsx]: damn, should I repush? (I just like how green the tests are - even selenium ones :D )

File: pkg/storaged/dialog.jsx (hunk 2 above)
Comment (KKoukiou) [reply pkg/storaged/dialog.jsx]: Sure, let's fix it separately, I wouldn't have realized it's broken if it was not for your PR anyway :)

**Pull request merged**